utils.go

```diff
@@ -0,0 +1,17 @@
+package engi
+
+import (
+	"crypto/rand"
```

Comment:
Yeah. `crypto/rand` is more *random*... But security isn't really what we need in this situation (speed is). `math/rand` should suffice.